components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/DefaultOIDCSessionStateManager.java

```diff
@@ -87,24 +88,60 @@ public Cookie addOPBrowserStateCookie(HttpServletResponse response) {
      * Adds the browser state cookie with tenant qualified path to the response.
      *
      * @param response
+     * @param request
      * @param tenantDomain
      * @return Cookie
      */
     @Override
-    public Cookie addOPBrowserStateCookie(HttpServletResponse response, String tenantDomain) {
+    public Cookie addOPBrowserStateCookie(HttpServletResponse response, HttpServletRequest request,
+                                          String tenantDomain) {
 
-        ServletCookie cookie = new ServletCookie(OIDCSessionConstants.OPBS_COOKIE_ID, UUID.randomUUID().toString());
-        cookie.setSecure(true);
+        ServletCookie cookie;
         if (IdentityTenantUtil.isTenantedSessionsEnabled()) {
+            // Invalidate the old opbs cookies which haven't tenanted paths.
+            removeOPBrowserStateCookiesInRoot(request, response);
+
+            cookie = new ServletCookie(OIDCSessionConstants.OPBS_COOKIE_ID, UUID.randomUUID().toString() +
+                    OIDCSessionConstants.TENANT_QUALIFIED_OPBS_COOKIE_SUFFIX);
+            cookie.setSecure(true);
             cookie.setPath(FrameworkConstants.TENANT_CONTEXT_PREFIX + tenantDomain + "/");
         } else {
+            cookie = new ServletCookie(OIDCSessionConstants.OPBS_COOKIE_ID, UUID.randomUUID().toString());
+            cookie.setSecure(true);
```

Comment:
This can be moved outside the if-else

Reply:
Addressed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1622/commits/a9bb2d43ee5c4d39ad85baedc5f91bb734c86761